sklearn/impute/tests/test_impute.py

```diff
@@ -1786,3 +1786,57 @@ def test_simple_imputer_constant_fill_value_casting():
         )
         X_trans = imputer.fit_transform(X_float32)
         assert X_trans.dtype == X_float32.dtype
+
+
+@pytest.mark.parametrize("strategy", ["mean", "median", "most_frequent", "constant"])
+def test_iterative_imputer_no_empty_features(strategy):
+    """Check the behavior of the iterative imputer.
+    When the input matrix contains no empty features, it should return
+    the same result regardless of whether `keep_empty_features`.
+    """
+    rng = np.random.RandomState(0)
+    X = _sparse_random_matrix(10, 10, density=0.95, random_state=rng).toarray()
+    X[X == 0] = np.nan
+
+    imputer_false = IterativeImputer(
```

Comment:
I think that we can do better naming instead of `imputer_false` and `imputer_true`: `imputer_keep_empty_features` and `imputer_drop_empty_features`.

Reply:
Addressed in https://github.com/scikit-learn/scikit-learn/pull/29779/commits/8a71a26d778593f383efa709ccc6ac569245df82